src/components/TextInput/TextInput.tsx

```diff
@@ -67,6 +71,8 @@ export interface InputProps {
       | React.ChangeEvent<HTMLInputElement>
       | React.ChangeEvent<HTMLTextAreaElement>
   ) => void;
+  /** The action to perform on the `input`/`textarea`'s onClick function  */
+  onClick?: (event: React.MouseEvent<HTMLInputElement, MouseEvent>) => void;
```

Comment:
Why `onClick` instead of `onFocus`?  This will not trigger if the input is accessed via the keyboard.

Reply:
I can add both.